.github/workflows/doctests.yml

```diff
@@ -0,0 +1,80 @@
+name: Doctests
+
+on:
+  push:
+    branches:
+      - doctest*
+  repository_dispatch:
+  schedule:
+    - cron: "0 0 * * *"
+
+env:
+  HF_HOME: /mnt/cache
+  RUN_SLOW: yes
+  OMP_NUM_THREADS: 16
+  MKL_NUM_THREADS: 16
+
+jobs:
+  run_doctests:
+    runs-on: [single-gpu, nvidia-gpu, a10, ci]
+    container:
+      image: huggingface/diffusers-all-latest-gpu
+      options: --gpus 0 --shm-size "16gb" --ipc host -v /mnt/cache/.cache/huggingface:/mnt/cache/
+
+    steps:
+      - name: Checkout diffusers
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+
+      - name: NVIDIA-SMI
+        uses: actions/checkout@v3
+        run: |
+          nvidia-smi
+
+      - name: Install dependencies
+        run: python3 -m pip install -e .[quality,test,training]
+
+      - name: Environment
+        run: |
+          python3 utils/print_env.py
+
+      - name: Get doctest files
+        run: |
+          $(python3 -c 'from utils.tests_fetcher import get_all_doctest_files; to_test = get_all_doctest_files(); to_test = " ".join(to_test); fp = open("doc_tests.txt", "w"); fp.write(to_test); fp.close()')
```

Comment:
Should it be `check_doctest_list` instead of `tests_fetcher`?

Reply:
From my understanding `check_doctest_list` is only a clean up function. The test fetcher is what actually fetches the relevant doc files to test when a change it made.

Reply:
Can't we use pre-built workflows to fetch the file changes? Example: https://github.com/huggingface/peft/blob/9119b780ebac7859db5753ebad50d94ba803c99c/.github/workflows/test-docker-build.yml#L21

Reply:
We fetch all python files from `src` and parse EXAMPLE_DOC_STRING to run the snippets. We also parse all the snippets in the markdown files in the docs. 

In this case we are not using test fetcher to dynamically select doc tests to run. We are running doc tests for all snippets once a day (while excluding certain doc pages because they will take too long).

Reply:
Okay that makes sense. Thanks for explaining.